### PR TITLE
[JAVA][native] - Fix #8027 - import the auto generated supporting JSON class only when discriminator is needed

### LINKED ISSUE #8027: [BUG][JAVA][native] Generated model java files contain import for non-existent invoker JSON class (supporting files were not generated)
#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [ ] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [x] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

##### Description

The generated Java model files contain an import for ...invoker.JSON but this class does not exist because option generateSupportingFiles=false
(Also, there is no reference to the JSON class in the body of the code so the import can safely be removed.)

##### openapi-generator version

5.0.0-SNAPSHOT git revision 4dc8d2a3514af15ac13624d4dc8e2e179749e5bd
This is a regression. This error does not occur in 4.3.1

##### OpenAPI declaration file content or url

```yaml
openapi: 3.0.1

info:
  title: Bug Report
  description: Demonstration of bug in openapi-generator 5.0.0-SNAPSHOT git revision 4dc8d2a3514af15ac13624d4dc8e2e179749e5bd
  version: 1

paths: {}

components:
  schemas:
    Pojo:
      type: object
      additionalProperties: false
      properties:
        Dummy:
          type: string
```

##### Generation Details

Config file:
```yaml
generatorName: java
outputDir: bug
library: native
inputSpec: bug.yaml
templateDir: modules/openapi-generator/src/main/resources/Java

apiPackage: bug.api
modelPackage: bug.model
invokerPackage: bug.invoker
packageName: bug.openapi

generateModelDocs: false
generateModelTests: false
generateApiTests: false
generateAliasAsModel: false
generateSupportingFiles: false
generateModelDocumentation: false
generateApiDocumentation: false
generateModels: true
generateApis: false

additionalProperties:
  java8: true
  dateLibrary: java8
  serializationLibrary: jackson
```

##### Steps to reproduce

Store the OpenApi declaration file pasted above into `bug.yaml`
Store the generator config file pasted above into `bug-config.yaml`
Run openapi-generator:
```bash
$ mkdir -p bug
$ bin/generate-samples.sh bug-config.yaml
```
##### Related issues/PRs

Somewhat similar but different: https://github.com/OpenAPITools/openapi-generator/issues/8016

##### Suggest a fix
The following line needs some decision logic so that it is only included if the supporting files have been generated:
```
modules/openapi-generator/src/main/resources/Java/libraries/native/model.mustache:49:import {{invokerPackage}}.JSON;
```
There are about 7 repetitions of this in other mustache files.

### PR DESCRIPTION
**Pull request by MosheElisha**
Import the auto generated supporting JSON class only when generateSupportingFiles is true.

Closes #8027 

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `5.1.x`, `6.0.x`
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

### CODE REVIEW

**Comment by MosheElisha** [comment]
@bgong-mdsol @bbdouglas @sreeshas @jfiala @lukoyanov @cbornet @jeff9finger @karismann @Zomzog @lwlee2608 @nmuesch mentioning the technical committee as requested and the last modifier of the template file. Please review. Thanks in advance.

**Comment by Zomzog** [comment]
The import looks unused for the simple models so it may be just removed.
But not il all_of where it is used. So even with `generateSupportingFiles=true` it must be added.

**Comment by MosheElisha** [comment]
Thanks, @Zomzog . If `generateSupportingFiles=false`, we can surely remove the import otherwise it won't compile, right?

If so, isn't removing the `import ...JSON` when `generateSupportingFiles=false` is simpler?

If `JSON` is used and `generateSupportingFiles=false` you will have a compilation issue and you will understand you need to change `generateSupportingFiles` to `true`.

**Comment by wing328** [comment]
cc @OpenAPITools/generator-core-team

**Comment by Zomzog** [comment]
> Thanks, @Zomzog . If `generateSupportingFiles=false`, we can surely remove the import otherwise it won't compile, right?
> 
> If so, isn't removing the `import ...JSON` when `generateSupportingFiles=false` is simpler?
> 
> If `JSON` is used and `generateSupportingFiles=false` you will have a compilation issue and you will understand you need to change `generateSupportingFiles` to `true`.

What I think is you can just remove it from the model.mustache without any condition.
But in other model templates like the anyof the import is used so you must keep it. If the user make the choice of skipping generateSupportingFiles he must provide JSON class in another way.

**Comment by MosheElisha** [comment]
@Zomzog OK. Thanks for the review! I'll remove the `import ...JSON` from `model.mustache` all together.

**Comment by MosheElisha** [comment]
@Zomzog `continuous-integration/drone/pr` job failed. Seems `JSON` is used in model classes sometimes.

```
/drone/src/samples/client/petstore/java/native/src/main/java/org/openapitools/client/model/Dog.java:[118,3] cannot find symbol
[ERROR]   symbol:   variable JSON
[ERROR]   location: class org.openapitools.client.model.Dog
```

For example, `openapi-generator/samples/client/petstore/java/native/src/main/java/org/openapitools/client/model/Dog.java`:

```
public class Dog extends Animal {
...
static {
  // Initialize and register the discriminator mappings.
  Map<String, Class<?>> mappings = new HashMap<String, Class<?>>();
  mappings.put("Dog", Dog.class);
  JSON.registerDiscriminator(Dog.class, "className", mappings);
}
}
```

**Comment by Zomzog** [comment]
Ok, in this case maybe the import condition may be done like 
```
{{#model}}
{{#discriminator}}
import {{invokerPackage}}.JSON;
{{/discriminator}}
```
I'm not sure it's really useful to try to remove this import because most of the supporting files must be generated or provided in one way or another. Does this one really create an issue in particular cases?

**Comment by MosheElisha** [comment]
Thanks, @Zomzog . That is a good observation. There is a `generateSupportingFiles` and you assume that it should be possible to make it false but as it turns out it the "supporting files" are mandatory in this case.

I'm guessing one approach is to make the supporting files list as minimal as possible. For example:

```
<supportingFilesToGenerate>JSON.java,RFC3339DateFormat.java</supportingFilesToGenerate>
```

In my case, I don't want and don't need the supporting files. I just want to generate model classes as POJOs.
The problem I have is that these two supporting files also require me to add dependencies on `jackson-datatype-jsr310`, `jackson-databind-nullable` and `jakarta.ws.rs-api`.

In my case, my auto generated model classes are not using the JSON class at all so I don't need the supporting files and all the dependencies. I want to be able to share and use the "model classes" in other use cases (e.g., sending them in a queue) and I want the receiving end to be able to depend on the "model classes" jar without too many unnecessary dependencies.

I will try your suggestion with the `discriminator` flag.

**Comment by MosheElisha** [comment]
@wing328 @Zomzog I believe `Shippable` failure is unrelated to my changes can you please verify/rerun and review if PR is OK?

I tested `discriminator` flag in the `pojo.mustache` as it is undefined in the `model.mustache`.

Thanks.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java [hunk 1]
```diff
@@ -233,6 +233,7 @@ void configureGeneratorProperties() {
 
         config.additionalProperties().put(CodegenConstants.GENERATE_APIS, generateApis);
         config.additionalProperties().put(CodegenConstants.GENERATE_MODELS, generateModels);
+        config.additionalProperties().put("generateSupportingFiles", generateSupportingFiles);
```
Comment (Zomzog) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java:236]: I think you can remove this line.

**Review by Zomzog — CHANGES_REQUESTED** [review]

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java (hunk 1 above)
Comment (MosheElisha) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultGenerator.java]: Indeed. Done.

**Comment by wing328** [comment]
Tested locally and no issue found.

I found some warnings though:
```
[WARNING] Javadoc Warnings
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/JSON.java:63: warning: no @return
[WARNING] public static Class<?> getClassForElement(JsonNode node, Class<?> modelClass) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/JSON.java:168: warning: no @param for visitedClasses
[WARNING] public static boolean isInstanceOf(Class<?> modelClass, Object inst, Set<Class<?>> visitedClasses) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/JSON.java:168: warning: no @return
[WARNING] public static boolean isInstanceOf(Class<?> modelClass, Object inst, Set<Class<?>> visitedClasses) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/FakeApi.java:84: warning: no @return
[WARNING] public CompletableFuture<Void> createXmlItem(XmlItem xmlItem) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/FakeApi.java:519: warning: no @return
[WARNING] public CompletableFuture<Void> testBodyWithFileSchema(FileSchemaTestClass body) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/FakeApi.java:599: warning: no @return
[WARNING] public CompletableFuture<Void> testBodyWithQueryParams(String query, User body) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/FakeApi.java:798: warning: no @return
[WARNING] public CompletableFuture<Void> testEndpointParameters(BigDecimal number, Double _double, String patternWithoutDelimiter, byte[] _byte, Integer integer, Integer int32, Long int64, Float _float, String string, File binary, LocalDate date, OffsetDateTime dateTime, String password, String paramCallback) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/FakeApi.java:903: warning: no @return
[WARNING] public CompletableFuture<Void> testEnumParameters(List<String> enumHeaderStringArray, String enumHeaderString, List<String> enumQueryStringArray, String enumQueryString, Integer enumQueryInteger, Double enumQueryDouble, List<String> enumFormStringArray, String enumFormString) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/FakeApi.java:997: warning: no @return
[WARNING] public CompletableFuture<Void> testGroupParameters(APItestGroupParametersRequest apiRequest) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/FakeApi.java:1035: warning: no @return
[WARNING] public CompletableFuture<Void> testGroupParameters(Integer requiredStringGroup, Boolean requiredBooleanGroup, Long requiredInt64Group, Integer stringGroup, Boolean booleanGroup, Long int64Group) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/FakeApi.java:1216: warning: no @return
[WARNING] public CompletableFuture<Void> testInlineAdditionalProperties(Map<String, String> param) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/FakeApi.java:1296: warning: no @return
[WARNING] public CompletableFuture<Void> testJsonFormData(String param, String param2) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/FakeApi.java:1378: warning: no @return
[WARNING] public CompletableFuture<Void> testQueryParameterCollectionFormat(List<String> pipe, List<String> ioutil, List<String> http, List<String> url, List<String> context) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/PetApi.java:79: warning: no @return
[WARNING] public CompletableFuture<Void> addPet(Pet body) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/PetApi.java:159: warning: no @return
[WARNING] public CompletableFuture<Void> deletePet(Long petId, String apiKey) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/PetApi.java:521: warning: no @return
[WARNING] public CompletableFuture<Void> updatePet(Pet body) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/PetApi.java:602: warning: no @return
[WARNING] public CompletableFuture<Void> updatePetWithForm(Long petId, String name, String status) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/StoreApi.java:76: warning: no @return
[WARNING] public CompletableFuture<Void> deleteOrder(String orderId) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/UserApi.java:76: warning: no @return
[WARNING] public CompletableFuture<Void> createUser(User body) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/UserApi.java:155: warning: no @return
[WARNING] public CompletableFuture<Void> createUsersWithArrayInput(List<User> body) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/UserApi.java:234: warning: no @return
[WARNING] public CompletableFuture<Void> createUsersWithListInput(List<User> body) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/UserApi.java:313: warning: no @return
[WARNING] public CompletableFuture<Void> deleteUser(String username) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/UserApi.java:577: warning: no @return
[WARNING] public CompletableFuture<Void> logoutUser() throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/api/UserApi.java:646: warning: no @return
[WARNING] public CompletableFuture<Void> updateUser(String username, User body) throws ApiException {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesAnyType.java:83: warning: no @param for key
[WARNING] public AdditionalPropertiesAnyType putAdditionalProperty(String key, Object value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesAnyType.java:83: warning: no @param for value
[WARNING] public AdditionalPropertiesAnyType putAdditionalProperty(String key, Object value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesAnyType.java:83: warning: no @return
[WARNING] public AdditionalPropertiesAnyType putAdditionalProperty(String key, Object value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesAnyType.java:95: warning: no @return
[WARNING] public Map<String, Object> getAdditionalProperties() {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesAnyType.java:102: warning: no @param for key
[WARNING] public Object getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesAnyType.java:102: warning: no @return
[WARNING] public Object getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesArray.java:84: warning: no @param for key
[WARNING] public AdditionalPropertiesArray putAdditionalProperty(String key, List value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesArray.java:84: warning: no @param for value
[WARNING] public AdditionalPropertiesArray putAdditionalProperty(String key, List value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesArray.java:84: warning: no @return
[WARNING] public AdditionalPropertiesArray putAdditionalProperty(String key, List value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesArray.java:96: warning: no @return
[WARNING] public Map<String, List> getAdditionalProperties() {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesArray.java:103: warning: no @param for key
[WARNING] public List getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesArray.java:103: warning: no @return
[WARNING] public List getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesBoolean.java:83: warning: no @param for key
[WARNING] public AdditionalPropertiesBoolean putAdditionalProperty(String key, Boolean value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesBoolean.java:83: warning: no @param for value
[WARNING] public AdditionalPropertiesBoolean putAdditionalProperty(String key, Boolean value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesBoolean.java:83: warning: no @return
[WARNING] public AdditionalPropertiesBoolean putAdditionalProperty(String key, Boolean value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesBoolean.java:95: warning: no @return
[WARNING] public Map<String, Boolean> getAdditionalProperties() {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesBoolean.java:102: warning: no @param for key
[WARNING] public Boolean getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesBoolean.java:102: warning: no @return
[WARNING] public Boolean getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesInteger.java:83: warning: no @param for key
[WARNING] public AdditionalPropertiesInteger putAdditionalProperty(String key, Integer value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesInteger.java:83: warning: no @param for value
[WARNING] public AdditionalPropertiesInteger putAdditionalProperty(String key, Integer value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesInteger.java:83: warning: no @return
[WARNING] public AdditionalPropertiesInteger putAdditionalProperty(String key, Integer value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesInteger.java:95: warning: no @return
[WARNING] public Map<String, Integer> getAdditionalProperties() {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesInteger.java:102: warning: no @param for key
[WARNING] public Integer getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesInteger.java:102: warning: no @return
[WARNING] public Integer getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesNumber.java:84: warning: no @param for key
[WARNING] public AdditionalPropertiesNumber putAdditionalProperty(String key, BigDecimal value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesNumber.java:84: warning: no @param for value
[WARNING] public AdditionalPropertiesNumber putAdditionalProperty(String key, BigDecimal value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesNumber.java:84: warning: no @return
[WARNING] public AdditionalPropertiesNumber putAdditionalProperty(String key, BigDecimal value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesNumber.java:96: warning: no @return
[WARNING] public Map<String, BigDecimal> getAdditionalProperties() {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesNumber.java:103: warning: no @param for key
[WARNING] public BigDecimal getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesNumber.java:103: warning: no @return
[WARNING] public BigDecimal getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesObject.java:83: warning: no @param for key
[WARNING] public AdditionalPropertiesObject putAdditionalProperty(String key, Map value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesObject.java:83: warning: no @param for value
[WARNING] public AdditionalPropertiesObject putAdditionalProperty(String key, Map value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesObject.java:83: warning: no @return
[WARNING] public AdditionalPropertiesObject putAdditionalProperty(String key, Map value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesObject.java:95: warning: no @return
[WARNING] public Map<String, Map> getAdditionalProperties() {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesObject.java:102: warning: no @param for key
[WARNING] public Map getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesObject.java:102: warning: no @return
[WARNING] public Map getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesString.java:83: warning: no @param for key
[WARNING] public AdditionalPropertiesString putAdditionalProperty(String key, String value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesString.java:83: warning: no @param for value
[WARNING] public AdditionalPropertiesString putAdditionalProperty(String key, String value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesString.java:83: warning: no @return
[WARNING] public AdditionalPropertiesString putAdditionalProperty(String key, String value) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesString.java:95: warning: no @return
[WARNING] public Map<String, String> getAdditionalProperties() {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesString.java:102: warning: no @param for key
[WARNING] public String getAdditionalProperty(String key) {
[WARNING] ^
[WARNING] /private/var/tmp/openapi-generator/samples/client/petstore/java/native-async/src/main/java/org/openapitools/client/model/AdditionalPropertiesString.java:102: warning: no @return
[WARNING] public String getAdditionalProperty(String key) {
[WARNING] ^
```
Will open an issue to track to see if anyone from the community can help fix these warnings.

**Comment by MosheElisha** [comment]
Thank you @wing328 and @Zomzog for reviewing!

**Comment by wing328** [comment]
@MosheElisha thanks for the PR, which has been included in the v5.1.0 release: https://twitter.com/oas_generator/status/1373636654024380423

**Pull request merged**